crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -395,35 +645,9 @@ impl_crypto_store! {
         // TODO: #2000 should make this lock go away, or change its shape.
         let _guard = self.save_changes_lock.lock().await;
 
-        let mut stores: Vec<&str> = [
-            (changes.private_identity.is_some() || changes.next_batch_token.is_some(), keys::CORE),
-            (changes.backup_decryption_key.is_some() || changes.backup_version.is_some(), keys::BACKUP_KEYS),
-            (!changes.sessions.is_empty(), keys::SESSION),
-            (
-                !changes.devices.new.is_empty()
-                    || !changes.devices.changed.is_empty()
-                    || !changes.devices.deleted.is_empty(),
-                keys::DEVICES,
-            ),
-            (
-                !changes.identities.new.is_empty() || !changes.identities.changed.is_empty(),
-                keys::IDENTITIES,
-            ),
-
-            (!changes.inbound_group_sessions.is_empty(), keys::INBOUND_GROUP_SESSIONS_V2),
-            (!changes.outbound_group_sessions.is_empty(), keys::OUTBOUND_GROUP_SESSIONS),
-            (!changes.message_hashes.is_empty(), keys::OLM_HASHES),
-            (!changes.withheld_session_info.is_empty(), keys::DIRECT_WITHHELD_INFO),
-            (!changes.room_settings.is_empty(), keys::ROOM_SETTINGS),
-            (!changes.secrets.is_empty(), keys::SECRETS_INBOX),
-        ]
-        .iter()
-        .filter_map(|(id, key)| if *id { Some(*key) } else { None })
-        .collect();
+        let indexeddb_changes = self.prepare_for_transaction(&changes).await?;
 
-        if !changes.key_requests.is_empty() {
-            stores.extend([keys::GOSSIP_REQUESTS])
-        }
+        let stores = indexeddb_changes.affected_stores();
```

Comment:
(light suggestion, feel free to dismiss) Would renaming `affected_stores` to `touched_stores` sound more natural?